app/src/org/commcare/android/framework/CommCareActivityUIController.java

```diff
@@ -0,0 +1,16 @@
+package org.commcare.android.framework;
+
+/**
+ * Interface to be implemented by any class acting as a ui controller for a CommCareActivity
+ *
+ * IMPORTANT: Any CommCareActivity that uses a CommCareActivityUIController must override
```

Comment:
Not thrilled about not having a better way of enforcing this -- if someone has an idea of a better way to set this up, let me know.